app/services/record_declaration.rb

```diff
@@ -0,0 +1,170 @@
+# frozen_string_literal: true
+
+class RecordDeclaration
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+
+  attribute :course_identifier
+  attribute :cpd_lead_provider
+  attribute :declaration_date, :datetime
+  attribute :declaration_type
+  attribute :participant_id
+  attribute :evidence_held
+
+  before_validation :declaration_attempt
+
+  validates :declaration_date, :declaration_type, presence: true
+  validates :declaration_date, future_date: true, declaration_date: true, allow_blank: true
+  validates :participant_id, participant_not_withdrawn: true
+  validates :cpd_lead_provider, induction_record: true
+  validates :course_identifier, course: true
+  validates :evidence_held,
+            presence: { message: I18n.t(:missing_evidence_held), if: :validate_evidence_held? },
+            inclusion: {
+              in: ->(record_declaration) { record_declaration.participant_profile.class::VALID_EVIDENCE_HELD },
```

Comment:
Could we ensure that a participant profile exists before running this validation? An example to this would be: If we leave out `course_identifier` from the params we get a 500 with the error: `NameError (uninitialized constant NilClass::VALID_EVIDENCE_HELD` where previously we would get `"The property '#/course_identifier' must be an available course to '#/participant_id'"`

Reply:
Fixed it https://github.com/DFE-Digital/early-careers-framework/pull/2225/files#diff-2fc4a0a9043f3924b5b01515d5102b99560a6498e3418b6abd82a7c985a7d36dR144 and added  test.